.gitpod.Dockerfile

```diff
@@ -0,0 +1,19 @@
+# Based on a file from https://github.com/mikenikles/cypress-on-gitpod
+# Licensed under the MIT license.
+
+FROM gitpod/workspace-full-vnc
+
+# Install Cypress dependencies.
+RUN sudo apt-get update \
+ && sudo DEBIAN_FRONTEND=noninteractive apt-get install -y \
+   libgtk2.0-0 \
+   libgtk-3-0 \
+   libnotify-dev \
+   libgconf-2-4 \
+   libnss3 \
+   libxss1 \
+   libasound2 \
+   libxtst6 \
+   xauth \
+   xvfb \
+ && sudo rm -rf /var/lib/apt/lists/*
```

Comment:
It definitely seems possible: https://www.gitpod.io/docs/config-docker#configure-a-public-docker-image

I would extend this `.gitpod.Dockerfile` from a Cypress Docker image instead then.

Reply:
The `.gitpod.Dockerfile` follows the same rules/syntax and has the same features as a "regular" Dockerfile.
So anything we can do in a normal Dockerfile, we can do here.

The build is really slow whenever I'm starting a new Gitpod container, so if we can speed this up by using the Cypress provided Docker images as a base, that would be great!

Not sure if the Cypress container images come with everything we need to have a visual display of the Cypress program on Gitpod though...

Reply:
I found this blog post, which has complicated instructions on getting the interactive mode going:

> If you want to see Cypress in interactive mode, you need to forward the XVFB messages from Cypress out of the Docker container into an X11 server running on the host machine. I have done this on my Mac; other operating systems might require different commands.

https://www.cypress.io/blog/2019/05/02/run-cypress-with-a-single-docker-command/#Interactive-mode

---

There's also a feature request for a built-in VNC server over on the Cypress Docker Github repo:
- https://github.com/cypress-io/cypress-docker-images/issues/468

---

I don't think I have a better or easier way to get this running based on the Cypress docker images, at least not a way that I feel competent adding or using safely.

Reply:
The lack of VNC support on the Cypress image is why I used `FROM gitpod/workspace-full-vnc` and added the Cypress dependencies.

I haven't looked into it since and there may be a better way.

An alternative may be to use Gitpod's VS Code Desktop support ([docs](https://www.gitpod.io/docs/develop/vscode-desktop-support)). In short, you start a Gitpod workspace in the cloud and connect to it from your local VS Code. With that, you should also be able to run the Cypress UI locally and won't need VNC in your `.gitpod.Dockerfile`.
It's worth noting I haven't tested this 😄. cc: @nisarhassan12 something for you to look into as well.

Reply:
> An alternative may be to use Gitpod's VS Code Desktop support ([docs](https://www.gitpod.io/docs/develop/vscode-desktop-support)). In short, you start a Gitpod workspace in the cloud and connect to it from your local VS Code. With that, you should also be able to run the Cypress UI locally and won't need VNC in your `.gitpod.Dockerfile`.

I'm not really sure what this means. If I understand correctly:

1. Install VS Code on my own computer.
1. Install Cypress on my computer as a binary.
1. Open Octoclairvoyant in Gitpod
1. Open dev server in Gitpod
1. Use local Cypress and VSCode to work with the dev server and the Gitpod files.

Is that correct?

Reply:
Yes, exactly. You keep the benefits of Gitpod prebuilds, isolated dev environments in the cloud but at the same time also get the benefits of running applications such as Cypress locally.

It's a potential alternative to VNC, which works too.

Reply:
I like being able to work from anywhere with Gitpod only. So I'll keep things as they are for now. Thanks for letting me know about the alternative way to configure things! 😄